### PR TITLE
Updated coprocessor registration to use ds param path, check hbase-si...

### PR DESCRIPTION
**Pull request by aheyne**
…te for registrations and register missing

### CODE REVIEW

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala [hunk 1]
```diff
@@ -65,17 +65,38 @@ trait HBaseFeatureIndex extends HBaseFeatureIndexType
     super.configure(sft, ds)
     val name = TableName.valueOf(getTableName(sft.getTypeName, ds))
     val admin = ds.connection.getAdmin
+    val coproUrl = ds.config.coprocessorUrl
+
+    def addCoprocessors(desc: HTableDescriptor, path: Path): Unit =
+      coprocessorList.foreach(c => addCoprocessor(c, desc, path))
+
+    def addCoprocessor(clazz: Class[_ <: Coprocessor], desc: HTableDescriptor, path: Path): Unit ={
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala]: (picky) Space after =.

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala [hunk 2]
```diff
@@ -65,17 +65,38 @@ trait HBaseFeatureIndex extends HBaseFeatureIndexType
...
-        // TODO: figure out if we need to discover this
-        val path = new Path(s"$rootDir/lib/geomesa-hbase-distributed-runtime.jar")
-        logger.info(s"Setting up coprocessors at $path")
-        // TODO: add all coprocessors
-        descriptor.addCoprocessor(classOf[KryoLazyDensityCoprocessor].getCanonicalName, path, Coprocessor.PRIORITY_USER, Maps.newHashMap[String, String]())
+        Option(admin.getConfiguration.get("hbase.coprocessor.user.region.classes")) match {
+          // if the coprocessors are installed site-wide don't register them in the table descriptor
+          case Some(value) =>
+            val installedCoprocessors = value.split(":").map(_.split(".").last).toSeq
+            coprocessorList.foreach{ c =>
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/index/HBaseFeatureIndex.scala]: (picky) space after 'foreach'

**Comment by jnh5y** [comment]
@aheyne could you be able to rebase on master and force push to clean things up a little bit?

File: geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/package.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/***********************************************************************
+ * Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ *************************************************************************/
```
Comment (jnh5y) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/package.scala]: This file doesn't seem to pass the mvn license:check:

[WARNING] Missing header in: /home/travis/build/locationtech/geomesa/geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/package.scala

**Comment by aheyne** [comment]
Closed for #1545

**Pull request closed without merging**